e2e/components/surveillance/complaints/complaints.po.js

```diff
@@ -1,21 +1,68 @@
-const elements = {
-  complaintsTable: '#complaints-table',
-  downloadResultsButton: '#download-results',
-};
-
 class ComplaintsComponent {
-  constructor () { }
+  constructor() {
+    this.elements = {
+      certificationBody: '#certification-body',
+      receivedDate: '#received-date',
+      acbComplaintId: '#acb-complaint-id',
+      complainantType: '#complainant-type',
+      summary: '#summary',
+      saveComplaint: '#action-bar-save',
+      closedDate: '#closed-date',
+      filter: '#data-filter',
+      downloadResultsButton: '#download-results',
+    };
+  }
+
+  set(fields) {
+    $(this.elements.certificationBody).click();
+    $(`//li[text()="${fields.body}"]`).click();
+    $(this.elements.receivedDate).addValue(fields.receivedDate);
+    $(this.elements.acbComplaintId).addValue(fields.acbId);
+    $(this.elements.summary).addValue(fields.summary);
+    $(this.elements.complainantType).click();
+    $(`//li[text()="${fields.type}"]`).click();
+  }
+
+  saveComplaint() {
+    return $(this.elements.saveComplaint).scrollAndClick();
+  }
+
+  get closedDate() {
+    return $(this.elements.closedDate);
+  }
+
+  fieldError(fieldName) {
+    return $(`#${fieldName}-helper-text`).getText();
+  }
+
+  getComplaintsTableHeaders() {
+    return $('table').$('thead').$$('th');
+  }
+
+  getComplaints() {
+    return $('table').$('tbody').$$('tr');
+  }
+
+  get downloadResultsButton() {
+    return $(this.elements.downloadResultsButton);
+  }
+
+  get filter() {
+    return $(this.elements.filter);
+  }
 
-  getComplaintsTableHeaders () {
-    return $(elements.complaintsTable).$('thead').$$('th');
+  getcellValue(row, col) {
```

Comment:
This doesn't look like the standard camelCase we use

```suggestion
  getCellValue(row, col) {
```

I'm also not sure it should be in this PO file at all. It looks like such a generic method that maybe it belongs in the hooks file. There's nothing about this method that cares what component/page it's on

Reply:
Moved table methods to hooks